pkg/ddc/goosefs/health_check.go

```diff
@@ -149,12 +153,16 @@ func (e *GooseFSEngine) checkMasterHealthy() (err error) {
 
 // checkWorkersHealthy check workers number changed
 func (e *GooseFSEngine) checkWorkersHealthy() (err error) {
-	workerName := e.getWorkerName()
-
 	// Check the status of workers
-	workers, err := kubeclient.GetStatefulSet(e.Client, workerName, e.namespace)
+	workers, err := ctrl.GetWorkersAsStatefulset(e.Client,
+		types.NamespacedName{Namespace: e.namespace, Name: e.getWorkerName()})
 	if err != nil {
-		return err
+		if fluiderrs.IsDeprecated(err) {
+			e.Log.Info("Warning: Deprecated mode is not support, so skip handling", "details", err)
```

Comment:
I suggest changing from `Warning: Deprecated mode is not support, so skip handling` to `Warning: The current dataset is created by dataset controller before v0.7.0, so scale out/in are not supported. To support these features, please create a new dataset.`

Reply:
Thanks for the suggestion. Should we also point out that "scale out/in are not supported" in `Events`? Or just in runtime controller's log?